paasta_tools/kubernetes/application/controller_wrappers.py

```diff
@@ -269,56 +269,43 @@ def update_related_api_objects(self, kube_client: KubeClient) -> None:
         super().update_related_api_objects(kube_client)
         self.sync_horizontal_pod_autoscaler(kube_client)
 
-    def should_have_hpa(self):
-        return (
-            (
-                self.soa_config.get_max_instances() is not None
-                or self.soa_config.config_dict.get("horizontal_autoscaling", False)
-            )
-            # with bespoke autoscaler, setup_kubernetes_job sets the number of instances directly; no HPA is required.
-            and self.soa_config.get_autoscaling_params()["decision_policy"] != "bespoke"
-            and self.soa_config.get_desired_state() != "stop"
-        )
-
     def sync_horizontal_pod_autoscaler(self, kube_client: KubeClient) -> None:
         """
         In order for autoscaling to work, there needs to be at least two configurations
         min_instnace, max_instance, and there cannot be instance.
         """
-        self.logging.info(
-            f"Syncing HPA setting for {self.item.metadata.name}/name in {self.item.metadata.namespace}"
-        )
-        hpa_exists = self.exists_hpa(kube_client)
-        if hpa_exists:
-            if not self.should_have_hpa():
-                # Remove HPA if autoscaling is disabled
-                self.delete_horizontal_pod_autoscaler(kube_client)
-                return
-            elif autoscaling_is_paused():
```

Comment:
Note I'm changing the behavior when autoscaling is paused.

Before we would set min_instances, so that when `self.soa_config.get_autoscaling_metric_spec` is called after that, the HPA would still be created, but not scale below the current instance count.

I think if we want to leave all the logic to the deploy config, we shouldn't have this interaction. Deleting the HPA will also have the effect of not changing the replica count. I think this is fine, based on this comment in the PR where pausing was implemented: https://github.com/Yelp/paasta/pull/2731#issuecomment-601401601

Reply:
Yeah I think that is cleaner -- it never really sat that well with me that we were editting the config class here.